### PR TITLE
Adonis-Cli Requires NPM Even When --yarn Flag Is On FIX

### PR DESCRIPTION
**Pull request by abrahamy**
This PR solves issue #39 . Instead of terminating with an error it gracefully terminates when npm is not installed.

The PR also added tests and travis/appveyor build scripts

### CODE REVIEW

**Comment by thetutlage** [comment]
Ohh that's a nice catch, but the build is failing, would you mind looking into it

File: src/Commands/Base.js [hunk 1]
```diff
@@ -43,9 +43,17 @@ class Base extends Command {
    */
   * _checkRequirements () {
     const nodeVersion = process.version
-    const npmVersion = yield pify(exec)('npm -v')
+    let npmVersion
     let error = false
 
+    // retrieve npm version.
+    pify(exec)('npm -v').then(version => {
```
Comment (RomainLanz) [inline src/Commands/Base.js]: I'd prefer to use `yield pify(...)` inside a try/catch block instead of using Promise directly here.

**Comment by abrahamy** [comment]
OK, I'll give it another shot and follow @RomainLanz's suggestion while at
it.

**Comment by RomainLanz** [comment]
Also it would be maybe better to check first if `npm` is installed (by the same way we check if `yarn` is installed https://github.com/adonisjs/adonis-cli/blob/develop/src/Commands/Base.js#L102) instead of doing that while trying to fetch the version.

**Comment by thetutlage** [comment]
I believe we are just doing too much here:

1. The simplest way to accept a parameter inside `checkRequirements` method on whether or not to validate npm version.
2. If user passes `--yarn` flag, then simply skip the npm check.

**Comment by abrahamy** [comment]
I already did that in a branch of my fork. I'd merge it to my PR. Please
have a look at it and let me know what you think.

**Comment by abrahamy** [comment]
@thetutlage @RomainLanz I have updated my PR and added tests and the build is no more failing. The reason the build was failing in the first instance was because `.travis.yml` and `appveyor.yml` build scripts were not in the repository. I have added these scripts and two tests to test for `--yarn` and `--npm` handling. Please kindly review and merge if all is as expected.

**Comment by thetutlage** [comment]
@abrahamy Going to check it no

**Comment by thetutlage** [comment]
Closed, since PR is for 3.x

**Pull request closed without merging**